GAD 8 - Binomial Heap/binomilia/HeapTester.java

```diff
@@ -0,0 +1,227 @@
+package gad.binomilia;
+
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * 
+ * @author Mihir Mahajan
+ * 
+ *         These tests only test if your BinomialHeap/BinomialTree itself works.
+ *         They don't test the logging. Still, they may not work if the logging
+ *         is incorrect!
+ * 
+ *         I can not guarantee the correctness or accuracy of these tests. I
+ *         hope they help, enjoy!
+ */
+
+public class HeapTester {
+
+	BinomialHeap heap = new BinomialHeap();
+	StudentResult result = new StudentResult();
+
+	@Test
+	void testIllegal() {
+		assertThrows(RuntimeException.class, () -> heap.min(), "Please throw an exception when heap is empty... thats the least u can do");
+		assertThrows(RuntimeException.class, () -> heap.deleteMin(result),"Would u please just throw me an exception >.<");
+	}
+
+	@Test
+	void testMin() {
+		for (int i = -300; i < 300; i++)
+			heap.insert(i, result);
+		assertEquals(-300, heap.min());
+	}
+
+	@Test
+	void testDeleteMin() {
+		List<Integer> nums = new ArrayList<>();
+		for (int i = -420; i < 420; i++) {
+			heap.insert(i, result);
+			nums.add(i);
+		}
+
+		for (int x : nums) {
+			assertEquals(x, heap.min(), "Wrong value for min!");
+			int deleted = heap.deleteMin(result);
+			assertEquals(x, deleted,
+					"Either did not delete the smallest number or the expected number wasnt in the heap");
+		}
+	}
+
+	@Test
+	void testBigNumbers() {
+		List<Integer> nums = new ArrayList<>();
+		for (int i = -420; i < 1000; i++) {

```

Comment:
The -420 (nice) makes the number of runs harder to count without any benefit. See also linea #139/#209/#218.